folsom-opentelemetry-metrics/src/main/java/com/spotify/folsom/client/OpenTelemetryMetrics.java

```diff
@@ -0,0 +1,192 @@
+
+    private static final Attributes HIT_TAGS = tag("result", "hit");
+    private static final Attributes MISS_TAGS = tag("result", "miss");
+    private static final Attributes FAILURE_TAGS = tag("result", "failure");
+    private static final Attributes SUCCESS_TAGS = tag("result", "success");
+
+    private final DoubleHistogram durations;
+    private final LongCounter gets;
+    private final LongCounter multigets;
+    private final LongCounter deletes;
+    private final LongCounter sets;
+    private final LongCounter incrDecrs;
+    private final LongCounter touches;
+
+    private final Set<OutstandingRequestsGauge> outstandingRequests = new CopyOnWriteArraySet<>();
+
+    /**
+     * Constructs an instance of {@link com.spotify.folsom.client.OpenTelemetryMetrics},
+     * initializing various metrics and gauges for monitoring Memcache-related activities using OpenTelemetry.
+     *
+     * @param meter The {@link io.opentelemetry.api.metrics.Meter} instance used to create and register metrics and gauges.
+     *              Metrics such as histograms and counters will be associated with specific Memcache
+     *              request types, while gauges will provide real-time observability for outstanding
+     *              requests and global connections.
+     */
+    public OpenTelemetryMetrics(final Meter meter) {
+        this.durations = meter.histogramBuilder("memcache.requests.nanoseconds").setUnit("nanoseconds").build();
+        this.gets = meter.counterBuilder("memcache.requests.get").build();
+        this.multigets = meter.counterBuilder("memcache.requests.multiget").build();
+        this.deletes = meter.counterBuilder("memcache.requests.delete").build();
+        this.sets = meter.counterBuilder("memcache.requests.set").build();
+        this.incrDecrs = meter.counterBuilder("memcache.requests.increment_decrement").build();
+        this.touches = meter.counterBuilder("memcache.requests.touch").build();
+
+        meter.gaugeBuilder("memcache.outstanding.requests").buildWithCallback(this::outstandingRequestsObservable);
+        meter.gaugeBuilder("memcache.global.connections").buildWithCallback(this::globalConnectionsObservable);
+    }
+
+    private void outstandingRequestsObservable(final ObservableDoubleMeasurement measurement) {
+        measurement.record(outstandingRequests
+                .stream()
+                .mapToLong(value -> (long) value.getOutstandingRequests()).sum());
+    }
+
+    private void globalConnectionsObservable(final ObservableDoubleMeasurement measurement) {
+        measurement.record(Utils.getGlobalConnectionCount());
+    }
+
+    @Override
+    public void measureGetFuture(final CompletionStage<GetResult<byte[]>> future) {
+        final long start = System.nanoTime();
+        future.whenComplete((result, throwable) -> {
+            final long duration = System.nanoTime() - start;
+            durations.record(duration, tag("operation", "get"));
```

Comment:
Nit: you can preallocate the `Attributes` instances, especially so when they don't contain any variables, just constants.